assets/js/amp-editor-blocks.js

```diff
@@ -211,6 +242,15 @@ var ampEditorBlocks = ( function() {
 			ampLayout = attributes.ampLayout;
 
 			if ( 'core/shortcode' === name ) {
+				// Lets remove amp-carousel from edit view.
+				if ( component.hasGalleryShortcodeCarouselAttribute( attributes.text || '' ) ) {
+					props.setAttributes( { text: component.removeAmpCarouselFromShortcodeAtts( attributes.text ) } );
```

Comment:
This will need to be moved to an `onChange` handler.

Reply:
Hmm, removing the attributes from the saved shortcode text on loading the block isn't related to any other attribute changing, the objective here was to not display the saved attributes within the shortcode already when the block is initially loaded, e.g. display just `[gallery]` instead of `[gallery amp-lightbox=true]`. Moving this part to `onChange` handler would cause the AMP attributes show up when first loading the block and remove these once `onChange` triggers.

Should we just display the attributes (in case they're not default) within the shortcode instead of trying to hide them? Thoughts?